src/Vector3.h

```diff
@@ -121,11 +126,7 @@ class Vector3
 	{
 		// Perform a bitwise comparison of the contents - we want to know whether this object is exactly equal
 		// To perform EPS-based comparison, use the EqualsEps() function
-		return (
-			(memcmp(&x, &a_Rhs.x, sizeof(x)) == 0) &&
-			(memcmp(&y, &a_Rhs.y, sizeof(y)) == 0) &&
-			(memcmp(&z, &a_Rhs.z, sizeof(z)) == 0)
-		);
+		return !((x != a_Rhs.x) || (y != a_Rhs.y) || (z != a_Rhs.z));
```

Comment:
You still have a template, which should work no matter what type it's used with. If someone uses it with a non-POD type it'll fail without warning.

IEEE-754 positive and negative zero are required to compare equal, which they do.
```
    std::cout << -0.0f << " == " << 0.0f <<std::endl;
    if(-0.0f == 0.0f)
    {
        std::cout << "true" << std::endl;
    }
    else
    {
        std::cout << "false" <<std::endl;
    }
```
Output:
```
-0 == 0
true
```